### PR TITLE
Add a plugin to shim the ES6 Promise implementation

### PR DESCRIPTION
**Pull request by clmath**
I am not sure how to test this plugin as the returned value will vary according to the tested platform.

### CODE REVIEW

**Comment by wkeese** [comment]
> I am not sure how to test this plugin as the returned value will vary according to the tested platform.

But, the returned value always has the same API, right?  It's not your job to test that the Promise shim (or the native implementation) work correctly, but you could have a simple smoke test unit test like:

```js
promiseAssertWorks: function () {
	var d = this.async(1000);
	new Promise(function (resolve) {
		setTimeout(resolve, 50);
	}).then(function () {
		d.resolve(true);
	});
},
```

The code itself looks fine to me.

**Comment by clmath** [comment]
I added the unit test and change the local require to the global require.

File: Promise.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Promise plugin.
+ *
+ * This plugin will return an ES6 compliant Promise implementation. This implementation come from the browser
```
Comment (cjolif) [inline Promise.js]: s/come/comes?

Comment (cjolif) [inline Promise.js]: will return->returns?

**Comment by clmath** [comment]
Merged in 89e7655a964b0f54e5c75b8cbe4ef4c2339a8e58

**Pull request closed without merging**